### PR TITLE
Checkboxes to hide quota and password

### PR DESCRIPTION
**Pull request by tomneedham**
Can now hide the password and quota fields using checkboxes and/or config options. Useful when configuring this screen for usage by group admins. 

screen shot 2018-06-11 at 12 21 05

Defaults to true as this is the current behaviour.

### CODE REVIEW

**Comment by PVince81** [comment]
@pmaier1 FYI

**Comment by pmaier1** [comment]
Well, not a huge change and I don't really see the use case. A little better overview maybe, yes. Anyway, nothing bad about it from my POV and provides additional flexibility. :+1:

**Comment by tomneedham** [comment]
Consistency was the driver

**Comment by tomneedham** [comment]
Rebased

**Comment by tomneedham** [comment]
@owncloud-bot rebase

**Comment by tomneedham** [comment]
@ownclouders rebase

**Comment by PVince81** [comment]
@tomneedham should we take over ? I think @sharidas could finish this

**Comment by mmattel** [comment]
Any update on this one?

**Comment by tomneedham** [comment]
Please take over

**Comment by sharidas** [comment]
I think we might have to add acceptance test for the changes made in this PR. Will try to add
- [ ] acceptance test to support this PR.

**Comment by phil-davis** [comment]
@sharidas I think there are acceptance tests that show/hide the email address. So hopefully that gives you a "template" for opening the settings menu and changing other settings. Ping us if you have questions.

**Comment by sharidas** [comment]
@phil-davis Yes, that's what I am moving with. Sure, in case of any queries I will come back to you.

**Comment by sharidas** [comment]
@phil-davis @individual-it I need your helping hands in reviewing the acceptance tests. 
I have verified the show/hide functionality manually in the UI.
Let me know if the acceptance test looks ok.

File: tests/acceptance/features/bootstrap/WebUIUsersContext.php [hunk 1]
```diff
@@ -383,6 +383,62 @@ public function theAdministratorShouldBeAbleToSeeEmailOfTheseUsers(TableNode $ta
...
+	/**
+	 * @Then /^the administrator should not be able to see the quota of these users in the User Management page:$/
+	 *
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 */
+	public function theAdministratorShouldNotBeAbleToSeeQuotaOfTheseUsers(TableNode $table) {
+		foreach ($table as $row) {
+			$userQuota = $this->usersPage->getQuotaOfUser($row['username']);
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIUsersContext.php]: Needs a method like `isQuotaColumnOfUserVisible`

File: tests/acceptance/features/webUISettingsMenu/settingsMenu.feature [hunk 2]
```diff
@@ -43,4 +43,62 @@ Feature: add users
...
+      | user1    |
+      | user2    |
+
+  Scenario: administrator should be able to see quota of user
+    When the administrator enables the setting "Show quota field" in the User Management page using the webUI
+    Then the administrator should be able to see the quota of these users in the User Management page:
+      | username |   quota   |
+      | user1    |   Default |
+      | user2    |   Default |
+    And the administrator changes the quota of user "user1" to "Unlimited" using the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUISettingsMenu/settingsMenu.feature]: We already have some test scenarios for changing quota.
I suggest remove the steps from here. Then make another scenario that has the 2 users and has their quota already set to "Unlimited" and "5 GB":
```
Given the administrator has changed the quota of user "user1" to "unlimited"
And the administrator has changed the quota of user "user2" to "5 GB"
```
and then do the action steps like:
```
When the administrator enables the setting "Show quota field" in the User Management page using the webUI
Then the administrator should be able to see the quota of these users in the User Management page:
  | username |   quota     |
  | user1    |   Unlimited |
  | user2    |   5 GB      |
```
That will check that displayed quota can show different values OK.

File: tests/acceptance/features/webUISettingsMenu/settingsMenu.feature [hunk 3]
```diff
@@ -43,4 +43,62 @@ Feature: add users
...
+      | user2    |   5 GB      |
+
+
+  Scenario: administrator should not be able to see quota of user
+    When the administrator disables the setting "Show quota field" in the User Management page using the webUI
+    Then the administrator should not be able to see the quota of these users in the User Management page:
+      | username |
+      | user1    |
+      | user2    |
+    And the administrator enables the setting "Show quota field" in the User Management page using the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUISettingsMenu/settingsMenu.feature]: We could delete the steps from here - I am not sure that they demonstrated anything more than the steps above.

**Comment by sharidas** [comment]
Not able to reproduce locally https://drone.owncloud.com/owncloud/user_management/564/51. Hence updating the branch to trigger CI again.

**Comment by sharidas** [comment]
@phil-davis Let me know your opinion/suggestion/feedback regarding the acceptance test updated.

**Comment by phil-davis** [comment]
Those "attempting to set an invalid quota" test scenarios have been failing intermittently. I was having a look last week - PR #141 which is just to help me get a bit more information. I raised issue #144 just now so that this problem is "officially recorded".

File: tests/acceptance/features/lib/UsersPage.php [hunk 4]
```diff
@@ -176,6 +178,55 @@ public function getEmailOfUser($username) {
...
+	 * @throws \Exception
+	 */
+	public function isQuotaColumnOfUserVisible($username) {
+		$userTr = $this->findUserInTable($username);
+		$userQuota = $userTr->find('xpath', $this->quotaColumnXpath);
+
+		if ($userQuota === null) {
+			throw new ElementNotFoundException(
+				__METHOD__ .
+				" xpath $this->passwordColumnXpath " .
```
Comment (phil-davis) [inline tests/acceptance/features/lib/UsersPage.php]: s/passwordColumnXpath/quotaColumnXpath

File: tests/acceptance/features/lib/UsersPage.php [hunk 5]
```diff
@@ -176,6 +178,55 @@ public function getEmailOfUser($username) {
...
+				"password column of user " . $username . " not found"
```
Comment (phil-davis) [inline tests/acceptance/features/lib/UsersPage.php]: s/password/quota

File: tests/acceptance/features/lib/UsersPage.php (hunk 4 above)
Comment (sharidas) [reply tests/acceptance/features/lib/UsersPage.php]: Done

File: tests/acceptance/features/lib/UsersPage.php (hunk 5 above)
Comment (sharidas) [reply tests/acceptance/features/lib/UsersPage.php]: Done

**Comment by sharidas** [comment]
Backport PR : https://github.com/owncloud/core/pull/34479

**Comment by PVince81** [comment]
as far as I remember that JS file is not unit testable

**Comment by mmattel** [comment]
Question, is this documentation relevant ?

**Comment by pmaier1** [comment]
> Question, is this documentation relevant ?

Maybe a screenshot of user management needs to be updated. See top post. In terms of functionality I don't think that this is worth documenting.

**Comment by mmattel** [comment]
@settermjd fyi. I am atm flat out.

**Pull request merged**